singular-kit/src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -0,0 +1,172 @@
+    private static final String API_SECRET = "secret";
+    private static final String DDL_TIME_OUT = "ddlTimeout";
+    String singularKey;
+    String singularSecret;
+    SingularConfig config;
+    long DDL_HANDLER_TIMEOUT_SEC = 60L;
+
+    Context context;
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        singularKey = settings.get(API_KEY);
+        singularSecret = settings.get(API_SECRET);
+        String ddlTimeout = settings.get(DDL_TIME_OUT);
+        if (!TextUtils.isEmpty(ddlTimeout)) {
+            try {
+                DDL_HANDLER_TIMEOUT_SEC = Long.parseLong(ddlTimeout);
+            } catch (Exception unableToGetDDLTimeout) {
+            }
+        }
+        this.context = context;
+        config = new SingularConfig(singularKey, singularSecret);
+        config.withDDLTimeoutInSec(DDL_HANDLER_TIMEOUT_SEC);
+        config.withDDLHandler(this);
+        Singular.init(context, config);
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean b) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "Singular";
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityResumed(Activity activity) {
+        Singular.onActivityResumed();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityPaused(Activity activity) {
+        Singular.onActivityPaused();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStopped(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivitySaveInstanceState(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityDestroyed(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> leaveBreadcrumb(String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logError(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logException(Exception e, Map<String, String> map, String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(MPEvent mpEvent) {
+        String eventName = mpEvent.getEventName();
+        Map eventInfo = mpEvent.getInfo();
+        if (eventInfo != null) {
+            JSONObject params = new JSONObject(eventInfo);
+            Singular.event(eventName, params.toString());
+        } else {
+            Singular.event(eventName);
+        }
+        return null;
```

Comment:
you should be returning a reporting message list here, see other kits for examples.